scraper/src/Scraper.hs

```diff
@@ -22,7 +23,45 @@ runScrape = do
 scrape :: IO ()
 scrape = do 
   website <- httpLBS "http://leg.colorado.gov/bill-search?field_sessions=10171&sort_bef_combine=field_bill_number%20ASC" 
-  L8.writeFile "test.html" $ getResponseBody website
+  L8.writeFile "test/example/bill_page_1.html" $ getResponseBody website
+
+-- I'm very sorry. Forgive me for my sins.
+fixNextPage = fmap mconcat . fmap mconcat $ fmap fromMaybe' nextPage
+fixLastPage = fmap mconcat . fmap mconcat $ fmap fromMaybe' lastPage
+
+-- This is far from ideal. please change it later 
+fromMaybe' mval = 
+  case mval of 
+    Nothing  -> undefined 
+    Just val -> val
+
+nextPage :: IO (Maybe [[ByteString]]) 
+nextPage = do 
+  website <- readFile "test/example/bill_page_1.html"
+  return $ S.scrapeStringLike website next
+
+    where 
+      next :: S.Scraper ByteString [[ByteString]]
+      next = S.chroots ("li" S.@: [S.hasClass "pager-next"]) link
+
+      link :: S.Scraper ByteString [ByteString]
+      link = S.chroots "a" $ do 
+        href <- S.attr "href" S.anySelector
+        return href
```

Comment:
This isn't exactly what you want. Another way to do this is to write:

```haskell
link :: S.Scraper ByteString ByteString
link = do
  href <- S.attr "href" "a"
  return href
```

Since Scalpel provides an `IsString` instance for selectors, a string literal will select elements matching the tag. Because we know there's only one, we can use this directly.

On a more general note, whenever you have the pattern:

```haskell
foo <- bar
return foo
```

you can replace it with

```
bar
```

due to the monad laws :)

Reply:
This is fixed.